subprojects/tooling-api/src/main/java/org/gradle/tooling/PhasedBuildActionExecuter.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling;
+
+import org.gradle.api.Incubating;
+import org.gradle.tooling.exceptions.MultipleBuildActionsException;
+
+/**
+ * Used to execute multiple {@link BuildAction}s in different phases of the build process.
+ *
+ * @since 4.7
+ */
+@Incubating
+public interface PhasedBuildActionExecuter extends ConfigurableLauncher<PhasedBuildActionExecuter> {
+
+    /**
+     * Builder for a {@link PhasedBuildActionExecuter}.
+     *
+     * <p>A single {@link BuildAction} is allowed per build phase. Use composite actions if needed.
+     *
+     * @since 4.7
+     */
+    @Incubating
+    interface Builder {
+
+        /**
+         * Adds the given action and its result handler to be executed after projects are loaded.
+         *
+         * <p>The action or model builders invoked by it are run when plugins have not yet be applied, and so it should try to get only gradle default models.
```

Comment:
⭕️ You can add model builders in an init script or with a settings plugin, so this comment is not accurate. It should say something along the lines of "models contributed by project plugins won't be available"